sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -38,6 +38,8 @@ public TransportFunctionalTests(bool isAsync) : base(isAsync)
                 new object[] { RequestContent.Create(new ReadOnlyMemory<byte>(new byte[10])), 10 },
                 new object[] { RequestContent.Create(new ReadOnlyMemory<byte>(new byte[10]).Slice(5)), 5 },
                 new object[] { RequestContent.Create(new ReadOnlySequence<byte>(new byte[10])), 10 },
+                new object[] { RequestContent.Create(new BinaryData("Hello, world")), 12 },
+                new object[] { RequestContent.Create(new BinaryData(new byte[10])), 10 },
             };
 
         [TestCaseSource(nameof(ContentWithLength))]
```

Comment:
Let's add a few Content tests to the https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/tests/HttpPipelineFunctionalTests.cs to make sure things work E2e

Reply:
Added some more asserts to existing tests around `ContentStream` and wrote a new one to ensure the correct thing happens  with `ExtractResponseContent()`